packages/visx-xychart/src/components/series/private/BaseBarGroup.tsx

```diff
@@ -0,0 +1,167 @@
+      dataKeys.forEach(key => {
+        const entry = dataRegistry.get(key);
+        if (entry && svgPoint && width && height && showTooltip) {
+          const datum = (horizontal ? findNearestDatumY : findNearestDatumX)({
+            point: svgPoint,
+            data: entry.data,
+            xScale,
+            yScale,
+            xAccessor: entry.xAccessor,
+            yAccessor: entry.yAccessor,
+            width,
+            height,
+          });
+          if (datum) {
+            showTooltip({
+              key,
+              ...datum,
+              svgPoint,
+            });
+          }
+        }
+      });
+    },
+    [dataKeys, dataRegistry, horizontal, xScale, yScale, width, height, showTooltip],
+  );
+  useEventEmitter('mousemove', handleMouseMove);
+  useEventEmitter('mouseout', hideTooltip);
+
+  const registryEntries = dataKeys.map(key => dataRegistry.get(key));
+
+  // if scales and data are not available in the registry, bail
+  if (registryEntries.some(entry => entry == null) || !xScale || !yScale || !colorScale) {
+    return null;
+  }
+
+  const [xMin, xMax] = xScale.range().map(Number);
+  const [yMax, yMin] = yScale.range().map(Number);
+
+  // try to figure out the 0 baseline for correct rendering of negative values
+  // we aren't sure if these are numeric scales or not ahead of time
+  const maybeXZero = xScale(0) ?? 0;
+  const maybeYZero = yScale(0) ?? 0;
+  const xZeroPosition = isValidNumber(maybeXZero)
+    ? // if maybeXZero _is_ a number, but the scale is not clamped and it's outside the domain
+      // fallback to the scale's minimum
+      Math.max(maybeXZero, Math.min(xMin, xMax))
+    : Math.min(xMin, xMax);
+  const yZeroPosition = isValidNumber(maybeYZero)
+    ? Math.min(maybeYZero, Math.max(yMin, yMax))
+    : Math.max(yMin, yMax);
+
+  const barThickness = getScaleBandwidth(groupScale);
+
+  const bars = registryEntries.flatMap(({ xAccessor, yAccessor, data, key }) =>
+    data.map((datum, index) => {
+      const barLength = horizontal
+        ? (xScale(xAccessor(datum)) ?? 0) - xZeroPosition
+        : (yScale(yAccessor(datum)) ?? 0) - yZeroPosition;
+
+      const groupPosition = (horizontal ? yScale(yAccessor(datum)) : xScale(xAccessor(datum))) ?? 0;
+
+      const withinGroupPosition = groupScale(key) ?? 0;
+
+      return {
+        key: `${key}-${index}`,
+        x: horizontal
+          ? xZeroPosition + Math.min(0, barLength)
+          : groupPosition + withinGroupPosition,
+        y: horizontal
+          ? groupPosition + withinGroupPosition
+          : yZeroPosition + Math.min(0, barLength),
+        width: horizontal ? Math.abs(barLength) : barThickness,
+        height: horizontal ? barThickness : Math.abs(barLength),
+        fill: colorScale(key),
+      };
+    }),
+  );
+
+  return (
+    <g className="visx-bar-group">
+      <BarsComponent bars={bars} horizontal={horizontal} xScale={xScale} yScale={yScale} />
```

Comment:
this is all the same as previous `BarGroup`, `bars` are just passed to `BarsComponent` now.